src/main/java/com/github/princesslana/eriscasper/ErisCasper.java

```diff
@@ -77,11 +78,42 @@ public static ErisCasper create() {
   }
 
   public static ErisCasper create(String token) {
-    return new ErisCasper(BotToken.of(token), Shard.fromConfig(CONFIG));
+
+    return new ErisCasper(BotToken.of(token), shardFromConfig(CONFIG));
   }
 
   public static ErisCasper create(String token, int shardNumber, int shardTotal) {
-    Shard shard = new Shard(shardNumber, shardTotal);
-    return new ErisCasper(BotToken.of(token), Optional.of(shard));
+    assertShard(shardNumber, shardTotal);
+    return new ErisCasper(
+        BotToken.of(token), Optional.of(ShardPayload.of(shardNumber, shardTotal)));
+  }
+
+  private static void assertShard(long shard, long total) {
+    Preconditions.checkArgument(shard >= 0, "Shard number must be greater than or equal to 0.");
+    Preconditions.checkArgument(total >= 1, "Shard total must be greater than or equal to 1.");
+    Preconditions.checkState(shard < total, "Shard number must be less than the shard total.");
+  }
+
+  public static Optional<ShardPayload> shardFromConfig(Properlty config) {
+    Optional<Integer> shard = config.getInt("ec.shard.id");
+    Optional<Integer> total = config.getInt("ec.shard.total");
+    if (shard.isPresent() || total.isPresent()) {
+      ErisCasperFatalException exception =
+          new ErisCasperFatalException(
+              "Failed to resolve both sharding values when only one was provided.");
+      return Optional.of(
+          shard
+              .map(
+                  s ->
+                      total
+                          .map(
+                              t -> {
+                                assertShard(s, t);
+                                return ShardPayload.of(s, t);
+                              })
+                          .orElseThrow(() -> exception))
+              .orElseThrow(() -> exception));
```

Comment:
```java
  public static Optional<ShardPayload> fromConfig(Properlty config) {
    Optional<Integer> shard = config.getInt("ec.shard.id");
    Optional<Integer> total = config.getInt("ec.shard.total");
    if (shard.isPresent() ^ total.isPresent()) {
      throw new ErisCasperFatalException(
          "Failed to resolve both sharding values when only one was provided.");
    }
    return shard.map(
        s -> {
          long t = total.get();
          check(s, t);
          return ShardPayload.of(s.longValue(), t);
        });
  }```
This would be a good alternate route if you want to take it.

Reply:
Or even
```java
  public static Optional<ShardPayload> fromConfig(Properlty config) {
    Optional<Long> shard = config.getLong("ec.shard.id");
    Optional<Long> total = config.getLong("ec.shard.total");
    if (shard.isPresent() ^ total.isPresent()) {
      throw new ErisCasperFatalException(
          "Failed to resolve both sharding values when only one was provided.");
    }
    return shard.map(
        s -> {
          long t = total.get();
          check(s, t);
          return ShardPayload.of(s, t);
        });
  }
```
to stop senseless long casting